#Download the dataset

In [3]:
!wget https://assets.supervisely.com/supervisely-supervisely-assets-public/teams_storage/6/O/RW/i7P4yZ7P2E4m2DuJBjHqxgl0AjZco5gwVzNUlADpFpfWgCHbzZVB94J2BGCNqFfh24xJVGgmmKIJsIdmWNbWIfgl0AzVishMvnpZ2UncKiIVqjA1isd4mV9QnHSh.tar

--2025-03-08 21:45:55--  https://assets.supervisely.com/supervisely-supervisely-assets-public/teams_storage/6/O/RW/i7P4yZ7P2E4m2DuJBjHqxgl0AjZco5gwVzNUlADpFpfWgCHbzZVB94J2BGCNqFfh24xJVGgmmKIJsIdmWNbWIfgl0AzVishMvnpZ2UncKiIVqjA1isd4mV9QnHSh.tar
Resolving assets.supervisely.com (assets.supervisely.com)... 94.46.171.158
Connecting to assets.supervisely.com (assets.supervisely.com)|94.46.171.158|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 667002880 (636M) [application/x-tar]
Saving to: ‘i7P4yZ7P2E4m2DuJBjHqxgl0AjZco5gwVzNUlADpFpfWgCHbzZVB94J2BGCNqFfh24xJVGgmmKIJsIdmWNbWIfgl0AzVishMvnpZ2UncKiIVqjA1isd4mV9QnHSh.tar’

i7P4yZ7P2E4m2DuJBjH 100%[===================>] 636.10M  31.2MB/s    in 23s     

2025-03-08 21:46:19 (27.3 MB/s) - ‘i7P4yZ7P2E4m2DuJBjHqxgl0AjZco5gwVzNUlADpFpfWgCHbzZVB94J2BGCNqFfh24xJVGgmmKIJsIdmWNbWIfgl0AzVishMvnpZ2UncKiIVqjA1isd4mV9QnHSh.tar’ saved [667002880/667002880]



In [5]:
!tar -xvf i7P4yZ7P2E4m2DuJBjHqxgl0AjZco5gwVzNUlADpFpfWgCHbzZVB94J2BGCNqFfh24xJVGgmmKIJsIdmWNbWIfgl0AzVishMvnpZ2UncKiIVqjA1isd4mV9QnHSh.tar -C dataset/

tar: Removing leading `/' from member names
/
LICENSE.md
README.md
augmented/
augmented/ann/
augmented/ann/D2S_99000004_D2S_augmented.jpg.json
augmented/ann/D2S_99000019_D2S_augmented.jpg.json
augmented/ann/D2S_99000052_D2S_augmented.jpg.json
augmented/ann/D2S_99000075_D2S_augmented.jpg.json
augmented/ann/D2S_99000117_D2S_augmented.jpg.json
augmented/ann/D2S_99000185_D2S_augmented.jpg.json
augmented/ann/D2S_99000192_D2S_augmented.jpg.json
augmented/ann/D2S_99000219_D2S_augmented.jpg.json
augmented/ann/D2S_99000241_D2S_augmented.jpg.json
augmented/ann/D2S_99000296_D2S_augmented.jpg.json
augmented/ann/D2S_99000306_D2S_augmented.jpg.json
augmented/ann/D2S_99000377_D2S_augmented.jpg.json
augmented/ann/D2S_99000385_D2S_augmented.jpg.json
augmented/ann/D2S_99000393_D2S_augmented.jpg.json
augmented/ann/D2S_99000409_D2S_augmented.jpg.json
augmented/ann/D2S_99000437_D2S_augmented.jpg.json
augmented/ann/D2S_99000448_D2S_augmented.jpg.json
augmented/ann/D2S_99000531_D2S_augmented.jpg.json
augment

In [6]:
!ls ./dataset/

augmented  LICENSE.md  meta.json  README.md  test  training  validation


# Data exploration

## Instance Categories

In [9]:
import json

In [10]:
metadata_path = "./dataset/meta.json"

with open(metadata_path, "r") as f:
  metadata = json.load(f)
metadata.keys()

dict_keys(['classes', 'tags', 'projectType'])

In [13]:
print("Number of object categories",len(metadata["classes"]))
print("Attributes of an instance",metadata["classes"][0].keys())


Number of object categories 60
Attributes of an instance dict_keys(['title', 'shape', 'color', 'geometry_config', 'id', 'hotkey'])


In [17]:
print("List of object categories:\n",*list(map(lambda instance: instance["title"],metadata["classes"])),sep="\n")

List of object categories:

adelholzener_alpenquelle_classic_075
adelholzener_alpenquelle_naturell_075
adelholzener_classic_bio_apfelschorle_02
adelholzener_classic_naturell_02
adelholzener_gourmet_mineralwasser_02
apple_braeburn_bundle
apple_golden_delicious
apple_granny_smith
apple_red_boskoop
augustiner_lagerbraeu_hell_05
augustiner_weissbier_05
avocado
banana_bundle
banana_single
cafe_wunderbar_espresso
caona_cocoa
carrot
clementine
clementine_single
coca_cola_05
coca_cola_light_05
cocoba_cocoa
corn_salad
corny_nussvoll
corny_nussvoll_single
corny_schoko_banane
corny_schoko_banane_single
cucumber
douwe_egberts_professional_ground_coffee
dr_oetker_vitalis_knuspermuesli_klassisch
ethiquable_gruener_tee_ceylon
franken_tafelreiniger
gepa_bio_caffe_crema
gepa_bio_und_fair_fencheltee
gepa_bio_und_fair_kamillentee
gepa_bio_und_fair_kraeuterteemischung
gepa_bio_und_fair_pfefferminztee
gepa_bio_und_fair_rooibostee
gepa_italienischer_bio_espresso
grapes_green_sugraone_seedless
grapes_sweet_c

In [20]:
# Check whether all images have the same size
# accross all sets
train_path = "./dataset/training/ann"
val_path = "./dataset/validation/ann"
test_path = "./dataset/test/ann"
aug_path = "./dataset/augmented/ann"

## Class Sizes

In [29]:
def check_image_size_variability(shapes):
  return all([shape == shapes[0] for shape in shapes[1:]])

def get_shapes(ann_path):
  shapes = []
  for root, subdirs, filenames in os.walk(ann_path):

    for filename in filenames:

      abs_path = os.path.join(root, filename)

      with open(abs_path, "r") as f:
        metadata = json.load(f)

      shapes.append((metadata['size']['height'],
                     metadata['size']['width']
                     ))
  return shapes

In [30]:
train_shapes = get_shapes(train_path)
aug_shapes = get_shapes(aug_path)
val_shapes = get_shapes(val_path)
test_shapes = get_shapes(test_path)

In [31]:
check_image_size_variability(train_shapes),check_image_size_variability(aug_shapes),check_image_size_variability(val_shapes),check_image_size_variability(test_shapes)

(True, True, True, True)

In [32]:
dataset_shapes = [*train_shapes,*aug_shapes,*val_shapes,*test_shapes]
check_image_size_variability(dataset_shapes)

True

In [33]:
train_shapes[0]

(1440, 1920)

All images in the dataset have the same shape of 1920*1440

## Class Balance

In [ ]:
# Very necessary

## Object Distribution

In [ ]:
# Nice to have not necessary